contracts/clients/src/BlsSigner.ts

```diff
@@ -67,20 +71,40 @@ export default class BlsSigner extends Signer {
       throw new TypeError("Transaction.to should be defined");
     }
 
+    const nonce = await BlsWalletWrapper.Nonce(
+      this.wallet.PublicKey(),
+      this.verificationGatewayAddress,
+      this.provider,
+    );
+
     // TODO: bls-wallet #375 Add multi-action transactions to BlsProvider & BlsSigner
     const action: ActionData = {
       ethValue: transaction.value?.toString() ?? "0",
       contractAddress: transaction.to.toString(),
       encodedFunction: transaction.data?.toString() ?? "0x",
     };
 
-    const nonce = await BlsWalletWrapper.Nonce(
-      this.wallet.PublicKey(),
-      this.verificationGatewayAddress,
+    const feeEstimate = await this.provider.estimateGas(transaction);
+
+    const aggregatorUtilitiesContract = AggregatorUtilities__factory.connect(
+      this.aggregatorUtilitiesAddress,
       this.provider,
     );
 
-    const bundle = this.wallet.sign({ nonce, actions: [action] });
+    const bundle = this.wallet.sign({
+      nonce,
+      actions: [
+        action,
+        {
+          ethValue: feeEstimate,
+          contractAddress: this.aggregatorUtilitiesAddress,
+          encodedFunction:
+            aggregatorUtilitiesContract.interface.encodeFunctionData(
+              "sendEthToTxOrigin",
+            ),
+        },
```

Comment:
`feeEstimate` didn't account for this extra action

Reply:
Fee estimation updated